### ISSUE TITLE
Azure Storage Queue Golang SDK leaks tcp connections on Linux

### ISSUE THREAD

**Issue by anhowe** [Azure/azure-storage-queue-go#8]
The queue golang SDK leaks TCP connections, and causes the program to run out of handles and stop running correctly.

To repro the leak:
1. add the below program to main.go
2. update storage account information with your SA, Key, and a queuename
3. run the program, and run `lsof | grep PROGRAMNAME | wc -l` and you will see the leaked TCP connections.
```
package main

import (
	"context"
	"fmt"
	"log"
	"net/url"
	"os"
	"os/signal"
	"time"

	"github.com/Azure/azure-storage-queue-go/2017-07-29/azqueue"
)

// built from https://github.com/Azure/azure-storage-queue-go/blob/master/2017-07-29/azqueue/zt_examples_test.go

func main() {
	storageAccount := "YOURSTORAGEACCOUNT"
	storageAccountKey := "YOURKEY"
	storageQueue := "queuename"
	credential := azqueue.NewSharedKeyCredential(storageAccount, storageAccountKey)
	p := azqueue.NewPipeline(credential, azqueue.PipelineOptions{})
	u, _ := url.Parse(fmt.Sprintf("https://%s.queue.core.windows.net", storageAccount))
	serviceURL := azqueue.NewServiceURL(*u, p)
	queueURL := serviceURL.NewQueueURL(storageQueue) // Queue names require lowercase
	messagesUrl := queueURL.NewMessagesURL()

	ctx := context.Background()
	log.Printf("started queuing messages")
	for i := 0; i < 100; i++ {
		messagesUrl.Enqueue(ctx, fmt.Sprintf("queuemsage%d", i), time.Duration(0)*time.Second, time.Duration(-1)*time.Second)
	}

	log.Printf("finished queuing messages")
	// wait on ctrl-c
	sigchan := make(chan os.Signal, 10)
	signal.Notify(sigchan)
	<-sigchan
	log.Printf("finished")
}
```

**Comment by anhowe** [comment]
This only repro's on linux, and I've been unable to get it to repro under Windows.

**Comment by anhowe** [comment]
The problem is with the queue's pipeline factory: https://github.com/Azure/azure-storage-queue-go/blob/master/2017-07-29/azqueue/zc_pipeline.go#L88.  Every request creates a "newDefaultHTTPClient()" until you send or receive just over 6000 queue messages and exhaust the limits.  The documentation for httpClient, https://golang.org/pkg/net/http/, says that "Clients and Transports are safe for concurrent use by multiple goroutines and for efficiency should only be created once and re-used.".

In contrast the azure pipeline code re-uses the same client as seen here https://github.com/Azure/azure-pipeline-go/blob/master/pipeline/core.go#L201.

**Comment by zezha-msft** [comment]
Hi @anhowe, I appreciate your patience. This should now be fixed in [0.2.0](https://github.com/Azure/azure-storage-queue-go/releases/tag/0.2.0).
